.github/workflows/build-test.yml

```diff
@@ -0,0 +1,26 @@
+name: Java CI
+
+on: [push]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 11
+        uses: actions/setup-java@v3
+        with:
+          java-version: '11'
+          distribution: 'adopt'
+      - name: Build with Maven
+        run: mvn --batch-mode --update-snapshots package
+      - uses: v1v/otel-upload-test-artifact-action@v2
```

Comment:
this action collects spans after the execution of the steps, it is fine, but maybe there is an implementation that uses [check_run](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#check_run) event trigger this would be more "real" span